api/v1alpha1/oidc_types.go

```diff
@@ -160,14 +159,12 @@ type OIDCProvider struct {
 }
 
 // OIDCDenyRedirect defines headers to match against the request to deny redirect to the OIDC Provider.
-// +notImplementedHide
 type OIDCDenyRedirect struct {
 	// Defines the headers to match against the request to deny redirect to the OIDC Provider.
 	Headers []OIDCDenyRedirectHeader `json:"headers"`
```

Comment:
Nit: while it's technically valid, an empty header array doesn't seem meaningful here.
Also, consider setting a max limit for the headers - most other APIs define one.
```suggestion
        // +kubebuilder:validation:MinItems=1
	// +kubebuilder:validation:MaxItems=16
	Headers []OIDCDenyRedirectHeader `json:"headers"`
```